projects/js-packages/publicize-components/src/components/form/connections-list.tsx

```diff
@@ -7,12 +9,25 @@ import styles from './styles.module.scss';
 import { useConnectionState } from './use-connection-state';
 
 export const ConnectionsList: React.FC = () => {
-	const { connections, toggleById } = useSocialMediaConnections();
+	const { recordEvent } = useAnalytics();
 
+	const { connections, toggleById } = useSocialMediaConnections();
 	const { canBeTurnedOn, shouldBeDisabled } = useConnectionState();
 
 	const { needsUserConnection } = usePublicizeConfig();
 
+	const toggleConnection = useCallback(
+		( connectionId: string, connection ) => () => {
+			toggleById( connectionId );
+			recordEvent( 'jetpack_social_connection_toggled', {
+				location: 'editor',
+				new_state: ! connection.enabled,
```

Comment:
At this point it's still the old state here

Reply:
While viewing analytics data, `new_state` will be confusing and vague and `{new_state: true}` or `{new_state: false}` won't give any meaningful information. It's better to keep it what it actually is in the connection data:

```suggestion
				enabled: ! connection.enabled,
```

Reply:
Originally I used enabled, but wanted to clarify that this state is the new state which it will be toggled to. But let's keep enabled then